### PR TITLE
proj + libgdal: update and rebuild dependants

### PR DESCRIPTION
**Pull request by ar-jan**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
    **I use PostGIS regularly and this works for me.**
- [x] I generally don't use the affected packages but briefly tested this PR
    **I generally don't use the other packages which have to be rebuilt against proj and/or gdal, but I did test all of them briefly**.

#### Motivation

- These updates are in preparation for adding QGIS.

### CODE REVIEW

**Comment by ar-jan** [comment]
Hello @Chocimier, do you have any insight into why postgis builds fail the checks in this PR? It's unrelated to the updates of gdal and proj, I see the same thing when just doing a compile locally on master branch. The build against updated gdal/proj (without checks) seems to function correctly locally.

Not sure if missing `xsltproc` is the actual problem, but adding `libxslt` to the template did not resolve that error.

File: srcpkgs/libgdal/template [hunk 1]
```diff
@@ -1,27 +1,63 @@
...
 fi
 
+do_build() {
+	: ${make_cmd:=make}
+	${make_cmd} ${makejobs} ${make_build_args} ${make_build_target}
+	# python modules
+	rm -f swig/python/*_wrap.cpp
+	${make_cmd} -C swig/python generate
+	cd swig/python
+	${make_cmd} ${makejobs} PYTHON=python3 ${makejobs} ${make_build_args} ${make_build_target}
```
Comment (Johnnynator) [inline srcpkgs/libgdal/template:50]: Imo add the last few lines as
```bash
post_build() {
	rm -f swig/python/*_wrap.cpp
	make -C swig/python generate
	cd swig/python
	make ${makejobs} PYTHON=python3 ${makejobs}
}
```
and do not define `do_build` at all

File: srcpkgs/libgdal/template [hunk 2]
```diff
@@ -1,27 +1,63 @@
...
+}
+
+do_install() {
+	: ${make_cmd:=make}
+	${make_cmd} DESTDIR=${DESTDIR} install
+	# python modules
+	cd swig/python
+	${make_cmd} PYTHON=python3 DESTDIR=${DESTDIR}/ install
```
Comment (Johnnynator) [inline srcpkgs/libgdal/template:58]: same as with `do_build` move the last 2 lines to `post_install` and do not overwrite `do_install` at all
```bash
	# python modules
	cd swig/python
	make PYTHON=python3 DESTDIR=${DESTDIR}/ install
```

File: srcpkgs/libgdal/template [hunk 3]
```diff
@@ -46,3 +82,17 @@ libgdal-devel_package() {
...
+python3-gdal_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - Python3 bindings"
+	pkg_install() {
+		vmove usr/bin/*.py
+		vmove "usr/lib/python*"
+		vlicense $wrksrc/$build_wrksrc/LICENSE.TXT
+		vdoc $wrksrc/$build_wrksrc/swig/python/README.rst
+		mkdir -p ${PKGDESTDIR}/usr/share/${pkgname}
+		cp -r $wrksrc/$build_wrksrc/swig/python/samples ${PKGDESTDIR}/usr/share/${pkgname}/examples
```
Comment (Johnnynator) [inline srcpkgs/libgdal/template:95]: `$wrksrc/$build_wrksrc` should be cwd, so no need to add it to the path (for all the above)

File: srcpkgs/libgdal/template [hunk 4]
```diff
@@ -46,3 +82,17 @@ libgdal-devel_package() {
...
+
+python3-gdal_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - Python3 bindings"
+	pkg_install() {
+		vmove usr/bin/*.py
+		vmove "usr/lib/python*"
+		vlicense $wrksrc/$build_wrksrc/LICENSE.TXT
+		vdoc $wrksrc/$build_wrksrc/swig/python/README.rst
+		mkdir -p ${PKGDESTDIR}/usr/share/${pkgname}
```
Comment (Johnnynator) [inline srcpkgs/libgdal/template:94]:
```suggestion
		vmkdir usr/share/python3-gdal
```
(using ${pkgname} isn't wrong, but writing out is imo more clear, and pkgnames usually don't change that often)q

File: srcpkgs/libgdal/template (hunk 3 above)
Comment (Johnnynator) [reply srcpkgs/libgdal/template:95]:
```suggestion
		vcopy swig/python/samples usr/share/python3-gdal/examples
```

File: srcpkgs/libkml/template [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+revision=1
+build_style=cmake
+makedepends="zlib-devel expat-devel boost-devel minizip-devel uriparser-devel"
+short_desc="Library to manipulate KML OGC files"
+maintainer="Nyx70 <n.y.x@bluewin.ch>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/libkml/libkml"
+distfiles="https://github.com/libkml/libkml/archive/${version}.tar.gz"
+checksum=8892439e5570091965aaffe30b08631fdf7ca7f81f6495b4648f0950d7ea7963
+patch_args=-Np1
```
Comment (Johnnynator) [inline srcpkgs/libkml/template:13]: `patch_args=-Np1` is the default, no need to specify.

File: srcpkgs/libgdal/template [hunk 6]
```diff
@@ -1,27 +1,63 @@
 # Template file for 'libgdal'
 pkgname=libgdal
-version=3.0.4
-revision=9
+version=3.2.3
+revision=1
+# aarch & arm temporary disabled
+archs="~aarch* ~armv*"
```
Comment (Johnnynator) [inline srcpkgs/libgdal/template:6]: Can you add a reason, or just why you disabled it.

File: srcpkgs/libgdal/template [hunk 7]
```diff
@@ -46,3 +82,17 @@ libgdal-devel_package() {
...
+	depends="${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - Python3 bindings"
+	pkg_install() {
+		vmove usr/bin/*.py
+		vmove "usr/lib/python*"
+		vlicense $wrksrc/$build_wrksrc/LICENSE.TXT
+		vdoc $wrksrc/$build_wrksrc/swig/python/README.rst
+		mkdir -p ${PKGDESTDIR}/usr/share/${pkgname}
+		cp -r $wrksrc/$build_wrksrc/swig/python/samples ${PKGDESTDIR}/usr/share/${pkgname}/examples
+		rm -rf ${PKGDESTDIR}/lib
```
Comment (Johnnynator) [inline srcpkgs/libgdal/template:96]: There should only be `/lib` if you would have created one in `pkg_install` but you didn't so this line should be a noop and be removed.

**Review by Johnnynator — CHANGES_REQUESTED** [review]
Did give it a quick review.

**Comment by Chocimier** [comment]
Disabled failing tests for now, please rebase.

**Comment by ar-jan** [comment]
I've changed/reverted the maintainer lines where applicable. I'd been in touch with Nyx70 previously and he gave thumbs up on continuing with his work, but wasn't explicit about maintainership.

Re `archs="~aarch* ~armv*"` for libgdal: I tried building for `armv7l`, and it failed with:

> x86_64-unknown-linux-gnu-gcc: error: unrecognized command-line option '-mfpu=vfpv3'
> x86_64-unknown-linux-gnu-gcc: error: unrecognized command-line option '-mfloat-abi=hard'
> error: command '/usr/bin/x86_64-unknown-linux-gnu-gcc' failed with exit code 1
> make: *** [GNUmakefile:73: build] Error 1
> [1m[31m=> ERROR: libgdal-3.2.3_1: post_build: 'make ${makejobs} PYTHON=python3 ${makejobs}' exited with 2
> [m[1m[31m=> ERROR:   in post_build() at srcpkgs/libgdal/template:45

Is this a matter of removing those flags, and if so, how to do that?

Finally there's one more package to be rebuilt against libgdal and proj, grass. The currently packaged version is apparently not working #29209; I built the new grass version 7.8.5 but it still has non-working GUI. So maybe that can be done separately.

I've resolved the xlint license issue for osg, but if I push that it'll trigger another build, maybe that's not desirable?

**Comment by sgn** [comment]
```diff
diff --git a/srcpkgs/libgdal/template b/srcpkgs/libgdal/template
index 2c6f38a328..f71d947b0a 100644
--- a/srcpkgs/libgdal/template
+++ b/srcpkgs/libgdal/template
@@ -2,8 +2,6 @@
 pkgname=libgdal
 version=3.2.3
 revision=1
-# aarch & arm currently failing
-archs="~aarch* ~armv*"
 wrksrc="gdal-${version}"
 build_style=gnu-configure
 configure_args="
@@ -39,20 +37,41 @@ build_options_default="kml"
 if [ -z "$CROSS_BUILD" ]; then
 	makedepends+=" hdf5-devel"
 fi
+CFLAGS="-pthread -I${XBPS_CROSS_BASE}/${py3_inc}"
+LDFLAGS="-L${XBPS_CROSS_BASE}/${py3_lib}"
 
 post_build() {
+	if [ "$CROSS_BUILD" ]; then
+		export PYPREFIX="$XBPS_CROSS_BASE"
+		export PYTHONPATH=${XBPS_CROSS_BASE}/${py3_lib}
+		for f in ${XBPS_CROSS_BASE}/${py3_lib}/_sysconfigdata_*; do
+			f=${f##*/}
+			export _PYTHON_SYSCONFIGDATA_NAME=${f%.py}
+		done
+	fi
+	export LDSHARED="${CC} $CFLAGS -shared $LDFLAGS"
+
 	rm -f swig/python/*_wrap.cpp
 	make -C swig/python generate
 	cd swig/python
-	make ${makejobs} PYTHON=python3 ${makejobs}
+	python3 setup.py build
 }
 
 post_install() {
 	vinstall gdal.pc 644 usr/lib/pkgconfig
 	vlicense LICENSE.TXT
-	# python modules
+	if [ "$CROSS_BUILD" ]; then
+		export PYPREFIX="$XBPS_CROSS_BASE"
+		export PYTHONPATH=${XBPS_CROSS_BASE}/${py3_lib}
+		for f in ${XBPS_CROSS_BASE}/${py3_lib}/_sysconfigdata_*; do
+			f=${f##*/}
+			export _PYTHON_SYSCONFIGDATA_NAME=${f%.py}
+		done
+	fi
+	export LDSHARED="${CC} $CFLAGS -shared $LDFLAGS"
+
 	cd swig/python
-	make PYTHON=python3 DESTDIR=${DESTDIR}/ install
+	python3 setup.py install --prefix=/usr --root=$DESTDIR
 }
 
 libgdal-tools_package() {
@@ -79,9 +98,8 @@ python3-gdal_package() {
 	depends="${sourcepkg}>=${version}_${revision}"
 	short_desc+=" - Python3 bindings"
 	pkg_install() {
-		vmove usr/bin/*.py
+		vmove "usr/bin/*.py"
 		vmove "usr/lib/python*"
-		vlicense LICENSE.TXT
 		vdoc swig/python/README.rst
 		vmkdir usr/share/python3-gdal
 		vcopy swig/python/samples usr/share/python3-gdal/examples
```

**Comment by ar-jan** [comment]
Thanks @sgn!

Re OpenOrienteering-Mapper, tests passed locally, it looks like the following failure is specific to the test runner. How best to handle that?

> 2021-06-30T02:21:26.4698109Z       Start 28: sensors_t
> 2021-06-30T02:21:26.9903615Z 28/28 Test #28: sensors_t ..........................***Failed    0.52 sec
> 2021-06-30T02:21:26.9905394Z QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
> 2021-06-30T02:21:26.9906423Z ********* Start testing of SensorsTest *********
> 2021-06-30T02:21:26.9908172Z Config: Using QtTest library 5.15.2, Qt 5.15.2 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 10.2.1 20201203), unknown unknown
> 2021-06-30T02:21:26.9909377Z PASS   : SensorsTest::initTestCase()
> 2021-06-30T02:21:26.9910507Z PASS   : SensorsTest::fakePositionSourcePluginTest()
> 2021-06-30T02:21:26.9911914Z PASS   : SensorsTest::fakePositionSourceSimulatedTest()
> 2021-06-30T02:21:26.9913326Z PASS   : SensorsTest::nmeaPositionSourcePluginTest()
> 2021-06-30T02:21:26.9914999Z QDEBUG : SensorsTest::nmeaPositionSourceSimulatedTest() NmeaPositionSource device: /dev/null
> 2021-06-30T02:21:26.9917801Z QDEBUG : SensorsTest::nmeaPositionSourceSimulatedTest() NmeaPositionSource device: /builddir/mapper-0.9.5/test/data/sensors/nmea.txt
> 2021-06-30T02:21:26.9920082Z QDEBUG : SensorsTest::nmeaPositionSourceSimulatedTest() NmeaPositionSource device: /tmp/sensors_t.EcPCyR
> 2021-06-30T02:21:26.9922022Z FAIL!  : SensorsTest::nmeaPositionSourceSimulatedTest() Compared values are not the same
> 2021-06-30T02:21:26.9923646Z    Actual   (int(source->error()))                    : 3
> 2021-06-30T02:21:26.9924385Z    Expected (int(QGeoPositionInfoSource::AccessError)): 0
> 2021-06-30T02:21:26.9925350Z    Loc: [../test/sensors_t.cpp(150)]
> 2021-06-30T02:21:26.9926306Z PASS   : SensorsTest::powershellPositionSourcePluginTest()
> 2021-06-30T02:21:26.9927534Z SKIP   : SensorsTest::powershellPositionSourceLiveTest() Powershell not available
> 2021-06-30T02:21:26.9928623Z    Loc: [../test/sensors_t.cpp(169)]
> 2021-06-30T02:21:26.9929544Z PASS   : SensorsTest::powershellPositionSourceSimulatedTest()
> 2021-06-30T02:21:26.9930561Z PASS   : SensorsTest::cleanupTestCase()
> 2021-06-30T02:21:26.9931272Z Totals: 7 passed, 1 failed, 1 skipped, 0 blacklisted, 504ms
> 2021-06-30T02:21:26.9931955Z ********* Finished testing of SensorsTest *********
> 2021-06-30T02:21:26.9932756Z CMake Error at sensors_t-RUN.cmake:35 (message):
> 2021-06-30T02:21:26.9933323Z   Test sensors_t failed: 1
> 2021-06-30T02:21:26.9949629Z 96% tests passed, 1 tests failed out of 28
> 2021-06-30T02:21:26.9950377Z Total Test time (real) =  19.33 sec
> 2021-06-30T02:21:26.9951106Z The following tests FAILED:
> 2021-06-30T02:21:26.9951982Z 	 28 - sensors_t (Failed)
> 2021-06-30T02:21:26.9952450Z Errors while running CTest
> 2021-06-30T02:21:26.9952979Z FAILED: CMakeFiles/test.util 
> 2021-06-30T02:21:26.9953967Z cd /builddir/mapper-0.9.5/build && /usr/bin/ctest --force-new-ctest-process
> 2021-06-30T02:21:26.9954852Z ninja: build stopped: subcommand failed.
> 2021-06-30T02:21:26.9956018Z => ERROR: OpenOrienteering-Mapper-0.9.5_2: do_check: '${make_cmd} ${make_check_args} ${make_check_target}' exited with 1
> 2021-06-30T02:21:26.9957145Z => ERROR:   in do_check() at common/build-style/cmake.sh:116

**Comment by Johnnynator** [comment]
You can add
```bash
do_check() {
	cd build
	ctest -E 'sensors_t'
}
```
which disables that specific test

**Comment by ar-jan** [comment]
Apparently, the Grass GUI currently packaged is not functional (#29209). The updated version compiles and starts successfully, but in my testing the GUI disappears as soon as the mouse pointer enters its area. I do not see any error message. Any clues how to find the cause? If not perhaps it can be merged anyway since the GUI is already not functional in the current state and otherwise the library versions are a problem.

**Comment by ar-jan** [comment]
The Grass GUI issue is due to to wxPython4 with Python 3.9 which segfaults.

https://github.com/OSGeo/grass/issues/1261
https://github.com/OSGeo/grass/issues/1123

This issue is fixed in wxPython4-4.1.1 which requires wxWidgets 3.1, but I assume it's not desirable to update to the 3.1 branch since it isn't stable like the current 3.0 series. I tried with the backported [patch](https://src.fedoraproject.org/rpms/python-wxpython4/c/f5471fb86aaae46a686b85c654fcbb98516355e6?branch=rawhide) to wxPython4 but still had the same issue.

**Comment by ericonr** [comment]
> This issue is fixed in wxPython4-4.1.1 which requires wxWidgets 3.1, but I assume it's not desirable to update to the 3.1 branch since it isn't stable like the current 3.0 series.

Not only that, it doesn't even work for all our applications...

Do you have any suggestions?

**Comment by ar-jan** [comment]
That first part was probably wrong -- what seems to be the underlying bug is fixed in that version, but I've tried with the backported patch which does work [for Fedora](https://github.com/OSGeo/grass/issues/1123#issuecomment-734248290), but without success. So either I'm doing something wrong in using the patched wxPython, or maybe something else is wrong (but that would be weird given that it fails in exactly the same way as described there).

**Comment by ericonr** [comment]
What steps did you follow?

**Comment by ar-jan** [comment]
I built `wxPython4` with a backport of the patch as in [this branch](https://github.com/ar-jan/void-packages/tree/gdal-proj-wxPython), and built `grass` 7.8.5.

Then installed wxPython4 and grass from this local build, then ran `grass78`.

I supposed it's only a runtime dependency, but to be sure I also built `wxPython` (v3.0) with the same patch backported, since wxPython-devel is used in building grass, then re-built grass. Same result.

I just switched to a new installation using i3 and I'm now seeing an additional error that I didn't see before: `"(wxgui.py:24024): Gtk-CRITICAL: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkSpinButton"` (maybe unrelated to the GUI crashing).

**Comment by ar-jan** [comment]
The grass package held things back six months ago, otherwise everything looked good.

I've given it another try, but it doesn't look hopeful: Python 3.10 + wxPython < 4.1 just doesn't seem to work. I've used grass-7.8.7RC1, which contains a workaround for a [wxPython issue](https://github.com/OSGeo/grass/issues/1971) as well as fixes related to Python 3.10.
I've also backported patches from wxPython 4.1 to 4.0.7 (following Fedora [here](https://src.fedoraproject.org/rpms/python-wxpython4/c/f5471fb86aaae46a686b85c654fcbb98516355e6?branch=rawhide), [here](https://src.fedoraproject.org/rpms/python-wxpython4/c/f55a17c2d3b336c82fb44f7816ec63219847f4cc?branch=rawhide), and [here](https://src.fedoraproject.org/rpms/python-wxpython4/c/1d5fd72dff63d62dd19b77e489d4115245d4d453?branch=rawhide))

With these changes the grass gui still doesn't work, and it looks like it is simply [not feasible with wxPython 4.0](https://github.com/OSGeo/grass/issues/2019). I've included the grass and wxPython4 commits hoping someone can sanity-check this.

If there's no progress with grass, I propose to either:

1. Include 7.8.7RC1 as is: the --text mode is functional (I think, I rarely use grass), and since the curently packaged grass gui doesn't work either it's not a regression afaict.
2. Remove the grass package.

In both cases the wxPython4 update can be left out -- unless these Python 3.10-related patches are helpful in other places?

**Comment by ar-jan** [comment]
Hi @sgn, earlier [you fixed](https://github.com/void-linux/void-packages/pull/31687#issuecomment-871569727) libgdal cross-compiling, with libgdal 3.2.3. Now ARM builds fail again, with the same gdal 3.2.3 as well as witih 3.3 and 3.4. It seems something has changed with the build process, any ideas?

**Comment by ar-jan** [comment]
Hi @Chocimier, do you know why the CI build still sees postgis-postgresql12 as a dependant of libgdal? It tries to install postgis-postgresql12 even though it was removed. Compile and install works locally.

**Comment by Chocimier** [comment]
Hi @ar-jan, `postgis-postgresql12` was removed as template but not removed from binary repositories, there is no automatic process to do that. Either ignore or ask on IRC for removal from binary repo.

**Comment by ar-jan** [comment]
Anyone have any ideas of why the crossbuild fails? @sgn maybe? It was working before. It seems that the swig build is now attempted in the do_build step, while it should be occurring in the custom post_build step.

**Comment by Chocimier** [comment]
I could take a look on cross compilation, but logs expired. Please rebase.

**Comment by ar-jan** [comment]
I'm currently without a Void system available, will rebase later. @sgn was also going to look at it (via #36816).

**Comment by ar-jan** [comment]
I updated the branch. @Chocimier or @sgn want to take a look at libgdal cross compilation?

(The other builds are failing because postgis-postgresql12 install is attempted).

File: srcpkgs/libgdal/template [hunk 8]
```diff
@@ -1,30 +1,96 @@
...
+post_build() {
+	if [ "$CROSS_BUILD" ]; then
+		export PYPREFIX="$XBPS_CROSS_BASE"
+		export PYTHONPATH=${XBPS_CROSS_BASE}/${py3_lib}
+		for f in ${XBPS_CROSS_BASE}/${py3_lib}/_sysconfigdata_*; do
+			f=${f##*/}
+			export _PYTHON_SYSCONFIGDATA_NAME=${f%.py}
+		done
+	fi
+	export LDSHARED="${CC} $CFLAGS -shared $LDFLAGS"
```
Comment (sgn) [inline srcpkgs/libgdal/template:71]: You can use `build_helper=python3` (or python?) and remove this block

File: srcpkgs/libgdal/template [hunk 9]
```diff
@@ -1,30 +1,96 @@
...
+ freexl-devel
+ geos-devel
+ jasper-devel
+ json-c-devel
+ libcurl-devel
+ libopenexr-devel
+ libopenjpeg2-devel
+ libpodofo-devel
+ libqhull-devel
+ libwebp-devel
```
Comment (sgn) [inline srcpkgs/libgdal/template:36]: Please not split the lines like this.

**Comment by ar-jan** [comment]
I've updated the libgdal template like so (not pushing to avoid needless 4 hour CI build):

<details>
  <summary>Updated libgdal-3.4.3 template</summary>

```
# Template file for 'libgdal'
pkgname=libgdal
version=3.4.3
revision=1
wrksrc="gdal-${version}"
build_style=gnu-configure
build_helper="python3 numpy"
configure_args="--with-expat=yes --with-hdf5=yes --with-kml=no
 --with-liblzma=yes --with-opencl=yes --with-pg=yes --with-podofo=yes
 --with-python=yes --with-spatialite=yes --with-sqlite3=yes --with-webp=yes
 --with-zstd=yes"
hostmakedepends="gettext-devel json-c-devel pkg-config python3-numpy swig"
makedepends="boost-devel expat-devel freexl-devel geos-devel jasper-devel
 json-c-devel libcurl-devel libopenexr-devel libopenjpeg2-devel libpodofo-devel
 libqhull-devel libwebp-devel libxml2-devel libzstd-devel netcdf-devel
 ocl-icd-devel opencl2-headers pcre2-devel proj-devel python3-devel
 sqlite-devel postgresql-libs-devel"
short_desc="Geospatial Data Abstraction Library"
maintainer="Orphaned <orphan@voidlinux.org>"
license="MIT"
homepage="https://gdal.org/"
distfiles="https://github.com/OSGeo/gdal/releases/download/v${version}/gdal-${version}.tar.gz"
checksum=7244962628c82921b4a4903dbb721c7222b7d82ed5928fef55a52f87f68ad1fe
subpackages="python3-gdal libgdal-devel libgdal-tools"

if [ -z "$CROSS_BUILD" ]; then
	makedepends+=" hdf5-devel"
fi

post_install() {
	vinstall gdal.pc 644 usr/lib/pkgconfig
	vlicense LICENSE.TXT
}

libgdal-tools_package() {
	depends="${sourcepkg}>=${version}_${revision}"
	short_desc+=" - tools"
	pkg_install() {
		vmove usr/bin
	}
}

libgdal-devel_package() {
	depends="${sourcepkg}>=${version}_${revision}"
	short_desc+=" - development files"
	pkg_install() {
		vmove usr/bin/gdal-config
		vmove usr/include
		vmove usr/lib/pkgconfig
		vmove usr/lib/*.a
		vmove usr/lib/*.so
	}
}

python3-gdal_package() {
	depends="${sourcepkg}>=${version}_${revision}"
	short_desc+=" - Python3 bindings"
	pkg_install() {
		vmove "usr/bin/*.py"
		vmove "usr/lib/python*"
		vdoc swig/python/README.rst
		vmkdir usr/share/python3-gdal
	}
}
```
</details>

But it looks like starting with `swig/python/setup.py` the crossbase is no longer used correctly:

> make[1]: Entering directory '/builddir/gdal-3.4.3/swig'
> for dir in python ; do (cd $dir; make build) || exit; done
> make[2]: Entering directory '/builddir/gdal-3.4.3/swig/python'
> ...
> python setup.py build
> WARNING: "CXX=/bin/sh /builddir/gdal-3.4.3/libtool --mode=compile --silent --tag=CXX armv7l-linux-gnueabihf-c++" was defined in the environment and contains more than one word. Unsetting it since that is incompatible of distutils
> WARNING: "CC=/bin/sh /builddir/gdal-3.4.3/libtool --mode=compile --silent --tag=CC armv7l-linux-gnueabihf-gcc -pthread -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   -I/usr/armv7l-linux-gnueabihf/usr/include/python3.10 -I/usr/armv7l-linux-gnueabihf/usr/include    -L/usr/armv7l-linux-gnueabihf/usr/lib/python3.10 -L/usr/armv7l-linux-gnueabihf/usr/lib" was defined in the environment and contains more than one word. Unsetting it since that is incompatible of distutils
> running build
> running build_py
> creating build
> creating build/lib.linux-x86_64-cpython-310
> ...
> creating build/temp.linux-x86_64-cpython-310/extensions
> ...
> gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard -g -I/usr/armv7l-linux-gnueabihf/usr/include -fdebug-prefix-map=/builddir/Python-3.10.8=. -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard -g -I/usr/armv7l-linux-gnueabihf/usr/include -fdebug-prefix-map=/builddir/Python-3.10.8=. -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard -I/usr/armv7l-linux-gnueabihf/usr/include/python3.10 -I/usr/armv7l-linux-gnueabihf/usr/include -I/usr/armv7l-linux-gnueabihf/usr/lib/python3.10/site-packages/numpy/core/include -I/usr/armv7l-linux-gnueabihf/usr/include -fdebug-prefix-map=/builddir/gdal-3.4.3=. -Wall -Wextra -Winit-self -Wunused-parameter -Wmissing-prototypes -Wmissing-declarations -Wformat -Werror=format-security -Wno-format-nonliteral -Wlogical-op -Wshadow -Wmissing-include-dirs -Werror=vla -Wdate-time -Wnull-dereference -Wduplicated-cond -Wfloat-conversion -DGNM_ENABLED -iquote /builddir/gdal-3.4.3/port -I/usr/armv7l-linux-gnueabihf/usr/include/openjpeg-2.5 -DGDAL_COMPILATION -fPIC -I../../port -I../../gcore -I../../alg -I../../ogr/ -I../../ogr/ogrsf_frmts -I../../gnm -I../../apps -I/usr/include/python3.10 -I/usr/lib/python3.10/site-packages/numpy/core/include -I/builddir/gdal-3.4.3/include -c extensions/ogr_wrap.cpp -o build/temp.linux-x86_64-cpython-310/extensions/ogr_wrap.o -I/builddir/gdal-3.4.3/port -I/builddir/gdal-3.4.3/gcore -I/builddir/gdal-3.4.3/alg -I/builddir/gdal-3.4.3/gnm -I/builddir/gdal-3.4.3/ogr -I/builddir/gdal-3.4.3/ogr/ogrsf_frmts -I/builddir/gdal-3.4.3/frmts/vrt -I/builddir/gdal-3.4.3/apps
> gcc: error: unrecognized command-line option '-mfpu=vfpv3'
> gcc: error: unrecognized command-line option '-mfloat-abi=hard'

I think this is because `CC` and `CXX` are unset? https://github.com/OSGeo/gdal/blob/v3.4.3/gdal/swig/python/setup.py#L22-L42. 

Is that right? How to fix?

**Comment by classabbyamp** [comment]
libgdal was updated in #40225, can you rebase for the proj update?

**Comment by ar-jan** [comment]
(easier to make a new branch)

**Pull request closed without merging**